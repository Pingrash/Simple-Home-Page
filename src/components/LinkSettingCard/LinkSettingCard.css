.card-container {
  height: 60px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border: 1px solid black;
  padding: 1rem;
  margin: 0.5rem 0;
}

.card-favicon {
  height: 25px;
  width: 25px;
  padding-left: 1rem;
}

.card-title {
  flex: 2;
  text-align: center;
  text-transform: capitalize;
}

.card-url {
  flex: 3;
  text-align: center;
}

.card-button {
  flex: 1;
  width: 70px;
}
