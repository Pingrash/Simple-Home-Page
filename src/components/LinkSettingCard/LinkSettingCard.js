import React from 'react';
import PropTypes from 'prop-types';

import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';

import { ArrowUpward, ArrowDownward } from '@material-ui/icons';

const useStyles = makeStyles(theme => ({
  cardContainer: {
    height: 60,
    minWidth: 500,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '1rem',
    margin: '0.5rem 0'
  },
  cardButton: {
    flex: 1
  },
  cardFavicon: {
    width: 28,
    paddingLeft: '1rem'
  },
  cardTitle: {
    flex: 2,
    textAlign: 'center',
    textTransform: 'capitalize'
  },
  cardUrl: {
    flex: 3,
    textAlign: 'center'
  },
  mainContainer: {
    display: 'flex'
  },
  arrowContainer: {
    display: 'flex',
    flexDirection: 'column',
    height: 60,
    margin: 'auto 1rem auto 0'
  },
  arrow: {
    flex: 1,
    height: 30
  }
}));

const LinkSettingCard = ({ link, deleteLink, shiftLinkCard }) => {
  const classes = useStyles();

  return (
    <div className={classes.mainContainer}>
      <div className={classes.arrowContainer}>
        <Button
          onClick={() => shiftLinkCard('up', link.name)}
          className={classes.arrow}
          data-test='shift-button'
        >
          <ArrowUpward />
        </Button>
        <Button
          onClick={() => shiftLinkCard('down', link.name)}
          className={classes.arrow}
          data-test='shift-button'
        >
          <ArrowDownward />
        </Button>
      </div>
      <Paper className={classes.cardContainer} data-test='container'>
        {link.favicon ? (
          <img
            src={link.favicon}
            alt='favicon'
            data-test='favicon'
            className={classes.cardFavicon}
          />
        ) : null}
        <Typography
          component='span'
          variant='h6'
          className={classes.cardTitle}
          data-test='title'
        >
          {link.name}
        </Typography>
        <Typography
          component='span'
          variant='body2'
          className={classes.cardUrl}
          data-test='url'
        >
          {link.link}
        </Typography>
        <Button
          className='card-button'
          color='secondary'
          data-test='delete-button'
          onClick={() => deleteLink(link.name)}
        >
          Delete
        </Button>
      </Paper>
    </div>
  );
};

LinkSettingCard.propTypes = {
  link: PropTypes.shape({
    link: PropTypes.string.isRequired,
    favicon: PropTypes.string,
    name: PropTypes.string.isRequired
  }),
  deleteLink: PropTypes.func.isRequired
};

export default LinkSettingCard;
