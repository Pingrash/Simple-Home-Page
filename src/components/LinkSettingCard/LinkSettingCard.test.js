import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { findByDataTest, checkProps } from '../../../Utilities';
import LinkSettingCard from './LinkSettingCard';

const testFunc = () => null;

const props = {
  link: {
    link: 'link1',
    favicon: 'favicon1',
    name: 'name1'
  },
  deleteLink: testFunc
};

const setup = () => {
  const component = shallow(<LinkSettingCard {...props} />);
  return component;
};

describe('LinkSettingCard renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a link title', () => {
    const title = findByDataTest(wrapper, 'title');
    expect(title.length).toBe(1);
  });

  it('Should render a favicon', () => {
    const favicon = findByDataTest(wrapper, 'favicon');
    expect(favicon.length).toBe(1);
  });

  it('Should render a delete button', () => {
    const button = findByDataTest(wrapper, 'delete-button');
    expect(button.length).toBe(1);
  });
});

describe('LinkSettingCard should have the correct props', () => {
  it('Should not have any prop errors', () => {
    const propsError = checkProps(LinkSettingCard, props);
    expect(propsError).toBeUndefined();
  });
});

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<LinkSettingCard {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
