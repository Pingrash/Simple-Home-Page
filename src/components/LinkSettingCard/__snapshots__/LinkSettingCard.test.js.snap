// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<div
  className="makeStyles-mainContainer-6"
>
  <div
    className="makeStyles-arrowContainer-7"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-arrow-8"
      data-test="shift-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-arrow-8"
      data-test="shift-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="MuiPaper-root makeStyles-cardContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
    data-test="container"
  >
    <img
      alt="favicon"
      className="makeStyles-cardFavicon-3"
      data-test="favicon"
      src="favicon1"
    />
    <span
      className="MuiTypography-root makeStyles-cardTitle-4 MuiTypography-h6"
      data-test="title"
    >
      name1
    </span>
    <span
      className="MuiTypography-root makeStyles-cardUrl-5 MuiTypography-body2"
      data-test="url"
    >
      link1
    </span>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text card-button MuiButton-textSecondary"
      data-test="delete-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Delete
      </span>
    </button>
  </div>
</div>
`;
